packages/ra-core/src/controller/list/useListParams.ts

```diff
@@ -77,24 +77,27 @@ export interface ListParams {
  * } = listParamsActions;
  */
 export const useListParams = ({
-    resource,
-    filterDefaultValues,
-    sort = defaultSort,
-    perPage = 10,
     debounce = 500,
     disableSyncWithLocation = false,
+    filterDefaultValues,
+    perPage = 10,
+    resource,
+    sort = defaultSort,
+    storeKey,
 }: ListParamsOptions): [Parameters, Modifiers] => {
+    if (!storeKey) storeKey = `${resource}.listParams`;
```

Comment:
This should raise a warning: you're modifying a function argument.

Please create another constant (like `finalStoreKey`) instead. 

```jsx
const finalStoreKey = storeKey ?? `${resource}.listParams`;
```